halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/FlotillaDeployment.java

```diff
@@ -42,26 +46,50 @@ public SpinnakerEndpoints getEndpoints() {
   private ProviderInterface<T> providerInterface;
   private AccountDeploymentDetails<T> deploymentDetails;
 
+  private void waitForServiceUp(SpinnakerService service) {
+    DaemonTaskHandler.log("Waiting for " + service.getArtifact().getName() + " to appear healthy");
+    RunningServiceDetails details = getServiceDetails(service);
+    while (details.getHealthy() == 0) {
+      try {
+        Thread.sleep(TimeUnit.SECONDS.toMillis(10));
+      } catch (InterruptedException ignored) {
+      }
+      details = getServiceDetails(service);
+    }
+  }
+
   @Override
   public DeployResult deploy(String spinnakerOutputPath) {
     SpinnakerEndpoints.Services services = getEndpoints().getServices();
-    DaemonTaskHandler.newStage("Deploying Spinnaker services and dependencies");
+    DaemonTaskHandler.newStage("Bootstrapping a minimal Spinnaker installation");
+    providerInterface.bootstrapSpinnaker(deploymentDetails, services);
+
+    waitForServiceUp(services.getRedisBootstrap());
+    waitForServiceUp(services.getOrcaBootstrap());
+    waitForServiceUp(services.getClouddriverBootstrap());
 
-    providerInterface.deployService(deploymentDetails, services.getRedis());
+    DaemonTaskHandler.newStage("Deploying remainder of Spinnaker services");
+    OrcaService.Orca orca = providerInterface.connectTo(deploymentDetails, services.getOrcaBootstrap());
+    providerInterface.ensureRedisIsRunning(deploymentDetails, services.getRedis());
+    providerInterface.deployService(deploymentDetails, orca, services.getClouddriver());
+    providerInterface.deployService(deploymentDetails, orca, services.getDeck());
 
-    providerInterface.deployService(deploymentDetails, services.getClouddriver());
-    providerInterface.deployService(deploymentDetails, services.getDeck());
-    providerInterface.deployService(deploymentDetails, services.getEcho());
-    providerInterface.deployService(deploymentDetails, services.getFiat());
-    providerInterface.deployService(deploymentDetails, services.getFront50());
-    providerInterface.deployService(deploymentDetails, services.getGate());
-    providerInterface.deployService(deploymentDetails, services.getIgor());
-    providerInterface.deployService(deploymentDetails, services.getOrca());
-    providerInterface.deployService(deploymentDetails, services.getRosco());
+    providerInterface.deployService(deploymentDetails, orca, services.getEcho());
+    // providerInterface.deployService(deploymentDetails, orca, services.getFiat());
```

Comment:
Nothing as far as I can tell. I pinged Travis earlier tonight and we'll look at it tomorrow